core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -4146,9 +4149,52 @@ private static List<Expression> args(RexCall call,
         if (argValueList.size() == 2) {
           return list.append(argValueList.get(1));
         }
+        if (argValueList.size() == 1) {
+          return list.append(Expressions.constant(Math.exp(1)));
+        }
+        // fall through
+      case "LN":
+        return list.append(Expressions.constant(Math.exp(1)));
+      case "LOG10":
+        return list.append(Expressions.constant(BigDecimal.TEN));
+      default:
+        throw new AssertionError("Operator not found: " + call.getOperator());
+      }
+    }
+  }
+
+  /** Implementor for the {@code LN}, {@code LOG}, and {@code LOG10} operators.
+   *
+   * <p>Handles all logarithm functions using log rules to determine the
+   * appropriate base (i.e. base e for LN).
+   */
+  private static class LogMSImplementor extends AbstractRexCallImplementor {
+    LogMSImplementor() {
+      super("logMS", NullPolicy.STRICT, true);
+    }
+
+    @Override Expression implementSafe(final RexToLixTranslator translator,
+        final RexCall call, final List<Expression> argValueList) {
+      return Expressions.call(BuiltInMethod.LOGMS.method, args(call, argValueList));
+    }
+
+    private static List<Expression> args(RexCall call,
```

Comment:
this looks duplicated can you point out what's different here?

Reply:
remove